DependencyInjection/DoctrineExtension.php

```diff
@@ -842,11 +842,17 @@ private function loadMessengerServices(ContainerBuilder $container) : void
             $container->removeDefinition('doctrine.orm.messenger.event_subscriber.doctrine_clear_entity_manager');
         }
 
+        $transportFactoryDefinition = $container->getDefinition('messenger.transport.doctrine.factory');
         if (! class_exists(DoctrineTransportFactory::class)) {
-            return;
+            // If symfony/messenger < 5.1
+            if (!class_exists(\Symfony\Component\Messenger\Transport\Doctrine\DoctrineTransportFactory::class)) {
+                // Dont add the tag
+                return;
+            } else {
```

Comment:
Because the if-case contains a `return`, the logic inside this `else` doesn't need to be in an else-block.

Reply:
Thank you!